src/lib/raft/raft.c

```diff
@@ -983,6 +1033,35 @@ raft_cfg_is_candidate(struct raft *raft, bool is_candidate)
 	raft_restore(raft);
 }
 
+void
+raft_notify_have_quorum(struct raft *raft, bool have_quorum)
+{
+	if (raft->have_quorum == have_quorum)
+		return;
+
+	raft->have_quorum = have_quorum;
+
+	raft_update(raft);
+}
+
+void
+raft_notify_leader_seen(struct raft *raft, bool leader_seen, uint32_t source)
```

Comment:
It really might look like a subjective thing, but I suggest to move the flag to the last argument. The source id feels like an object here and the flag is like an argument of its method. But you can say also tell me I am mad and leave it, and I won't object.

Reply:
Fair. Although I would change `raft_process_msg` too. Maybe someday later along with some other change.